overlay_minimal/etc/init.d/S35wifi

```diff
@@ -10,10 +10,17 @@ WIFI_MODULE=8189fs
 
 case "$1" in
   start)
-
 	logger -s "Starting wifi using module $WIFI_MODULE" 
 
-	modprobe $WIFI_MODULE
+	# Try to get the Wyze MAC address
+	WIFI_OPTIONS=""
+	if [[ "$USE_WYZE_MAC" -eq "1" ]] && [[ -f "/params/config/.product_config" ]]; then
+		MAC=$(awk -F= '/NETRELATED_MAC/ {print substr($2, 1, 12)}' /params/config/.product_config | sed 's/\(..\)/\1:/g;s/:$//')
```

Comment:
For us mere mortals can you provide a quick explanation of this witchcraft?

Reply:
Sure! On the line containing `NETRELATED_MAC`, `awk` will print the first 12 characters on the left side of the `=` delimiter. This will transform
```
NETRELATED_MAC=aabbccddeeffgghh...
```
into
```
aabbccddeeff
```

`sed` will then insert a `:` after each pair or characters then remove the trailing `:`. In short, it converts the previous example into
```
aa:bb:cc:dd:ee:ff
```